### ISSUE TITLE
Empty select field - ACF 5.6.x (Uncaught TypeError: Cannot read property 'slice' of undefined)

### ISSUE THREAD

**Issue by breadadams** [jonathan-dejong/acf-sites#6]
I've been using this plugin on a site for a while now and worked fine for the most part. However, since recently updating ACF to `5.6.x`, it seems that acf-sites has stopped working.

![Console error](https://user-images.githubusercontent.com/5795227/29209942-70cc4f5c-7e91-11e7-924d-6b1d45c8143d.png)

![Empty select field](https://user-images.githubusercontent.com/5795227/29209949-7b420314-7e91-11e7-8dc3-9caf80c6a434.png)

Any ideas @jonathan-dejong?

**Edit**: After some further testing it appears to just be an issue with select2 - as the radio/checkbox options still work fine.

**Comment by artuze** [comment]
It looks like all dropdowns ultimately generated by ACF's `acf_field_select->render_field()` add the data-ajax attribute, no matter what it is, then I guess remove it with JS before ultimately calling `select2()` on the element.

So since this plugin is trying to force it to not use AJAX anyway, maybe add in a `$select.removeAttr('data-ajax');` just before the `$select.select2()` call in the JS file? It worked for me.

**Comment by AbeCole** [comment]
Can confirm @artuze solution resolved the issue for me. Add `$select.removeAttr('data-ajax');` on line 15 to `acf-sites-acf-sites/assets/js/input.js`

**Comment by jonathan-dejong** [comment]
:D Wow I did not even know there was still usage for this plugin. 

I think it should get a complete rewrite to be properly functional with latest ACF tbh.. But I'm glad you guys found a workaround in the meantime!

**Comment by AbeCole** [comment]
Ye there is still usage since ACF don't offer a field like this natively. The other option would be to use action hooks to alter one of the Relational field types (i.e. Post Object, Relationship) to show the sites as options, but that isn't as intuitive or clear as having a specific field type for this functionality. So nice one for writing all that time ago, maybe you could make this one small update and leave it for a while! (Would do it myself but just gone away on holiday so will make a pull request if I remember next week)

**Comment by isuke01** [comment]
Yep, thanks for plugin, helped me a lot (didn't have to write it by myself :P) 
Just fixed this above select2 thing and everything works fine!
